File: app/src/main/java/io/plan8/backoffice/fragment/BaseFragment.java
package io.plan8.backoffice.fragment;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * Created by SSozi on 2017. 11. 28..
 */

public abstract class BaseFragment extends Fragment {
    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        Log.d("lifeCycle", "Fragment onAttach :: " + getClass().getSimpleName() + "  ::  " + hashCode());
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.d("lifeCycle", "Fragment onCreate :: " + getClass().getSimpleName() + "  ::  " + hashCode());
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        Log.d("lifeCycle", "Fragment onCreateView :: " + getClass().getSimpleName() + "  ::  " + hashCode());
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        Log.d("lifeCycle", "Fragment onActivityCreated :: " + getClass().getSimpleName() + "  ::  " + hashCode());
    }

    @Override
    public void onStart() {
        super.onStart();
        Log.d("lifeCycle", "Fragment onStart :: " + getClass().getSimpleName() + "  ::  " + hashCode());
    }

    @Override
    public void onResume() {
        super.onResume();
        Log.d("lifeCycle", "Fragment onResume :: " + getClass().getSimpleName() + "  ::  " + hashCode());
    }

    @Override
    public void onPause() {
        super.onPause();
        Log.d("lifeCycle", "Fragment onPause :: " + getClass().getSimpleName() + "  ::  " + hashCode());
    }

    @Override
    public void onStop() {
        super.onStop();
        Log.d("lifeCycle", "Fragment onStop :: " + getClass().getSimpleName() + "  ::  " + hashCode());
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        Log.d("lifeCycle", "Fragment onDestroyView :: " + getClass().getSimpleName() + "  ::  " + hashCode());
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Log.d("lifeCycle", "Fragment onDestroy :: " + getClass().getSimpleName() + "  ::  " + hashCode());
    }

    @Override
    public void onDetach() {
        super.onDetach();
        Log.d("lifeCycle", "Fragment onDetach :: " + getClass().getSimpleName() + "  ::  " + hashCode());
    }
}

File: app/src/main/java/io/plan8/backoffice/fragment/BaseFragment.kt
package io.plan8.backoffice.fragment

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

/**
 * Created by chokwanghwan on 2017. 11. 9..
 */
open class BaseFragment:Fragment() {
    override fun onAttach(context: Context?) {
        super.onAttach(context)
        Log.d("lifeCycle", "Fragment onAttach :: " + javaClass.simpleName + "  ::  " + hashCode())
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d("lifeCycle", "Fragment onCreate :: " + javaClass.simpleName + "  ::  " + hashCode())
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        Log.d("lifeCycle", "Fragment onCreateView :: " + javaClass.simpleName + "  ::  " + hashCode())
        return super.onCreateView(inflater, container, savedInstanceState)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        Log.d("lifeCycle", "Fragment onActivityCreated :: " + javaClass.simpleName + "  ::  " + hashCode())
    }

    override fun onStart() {
        super.onStart()
        Log.d("lifeCycle", "Fragment onStart :: " + javaClass.simpleName + "  ::  " + hashCode())
    }

    override fun onResume() {
        super.onResume()
        Log.d("lifeCycle", "Fragment onResume :: " + javaClass.simpleName + "  ::  " + hashCode())
    }

    override fun onPause() {
        super.onPause()
        Log.d("lifeCycle", "Fragment onPause :: " + javaClass.simpleName + "  ::  " + hashCode())
    }

    override fun onStop() {
        super.onStop()
        Log.d("lifeCycle", "Fragment onStop :: " + javaClass.simpleName + "  ::  " + hashCode())
    }

    override fun onDestroyView() {
        super.onDestroyView()
        Log.d("lifeCycle", "Fragment onDestroyView :: " + javaClass.simpleName + "  ::  " + hashCode())
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("lifeCycle", "Fragment onDestroy :: " + javaClass.simpleName + "  ::  " + hashCode())
    }

    override fun onDetach() {
        super.onDetach()
        Log.d("lifeCycle", "Fragment onDetach :: " + javaClass.simpleName + "  ::  " + hashCode())
    }
}